ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OTreeBasedSupervisedAlgorithm.scala

```diff
@@ -18,17 +18,18 @@
 package ai.h2o.sparkling.ml.algos
 
 import ai.h2o.sparkling.ml.models.H2OTreeBasedSupervisedMOJOModel
-import ai.h2o.sparkling.ml.params.H2OTreeBasedSupervisedMOJOParams
 import hex.Model
 import org.apache.spark.sql.Dataset
 
 import scala.reflect.ClassTag
 
-abstract class H2OTreeBasedSupervisedAlgorithm[P <: Model.Parameters: ClassTag]
-  extends H2OSupervisedAlgorithm[P]
-  with H2OTreeBasedSupervisedMOJOParams {
+abstract class H2OTreeBasedSupervisedAlgorithm[P <: Model.Parameters: ClassTag] extends H2OSupervisedAlgorithm[P] {
 
   override def fit(dataset: Dataset[_]): H2OTreeBasedSupervisedMOJOModel = {
     super.fit(dataset).asInstanceOf[H2OTreeBasedSupervisedMOJOModel]
   }
+
+  def getNtrees(): Int
```

Comment:
Do I understand correctly that the `getNTrees` and the setter are here because of MOJO so they are available there? Otherwise for the wrappers itself they are auto-generated, right?

Reply:
Yes, you're right NTrees are auto-generated. These methods are here just for ensuring API parity between`H2OTreeBasedSupervisedAlgorithm` and `H2OTreeBasedSupervisedMOJOModel`. You can see usege of these methods in `H2OTreeBasedSupervisedMOJOModelTestSuite.testTrainedTreeBasedModelHasPositiveNumberOfTrees`